### PR TITLE
www-apps/webdavcgi: Version bump to 1.1.1

### PR DESCRIPTION
**Pull request by paraenggu**
New version of www-apps/webdavcgi ebuild including the required dev-perl/DateTime-Format-Human-Duration dependency.

Gentoo-Bug: https://bugs.gentoo.org/595060

@dev-zero could you please review and merge? Thanks.

### CODE REVIEW

**Comment by gktrk** [comment]
@gentoo/proxy-maint

**Comment by paraenggu** [comment]
I had a quick chat with @dev-zero it seems that he's quite busy these days and doesn't find the time to review the PR. @gktrk is there someone else from gentoo which could step in and review the pull request?

File: www-apps/webdavcgi/webdavcgi-1.1.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	ewarn "common group, and allow access to the following directories by this"
+	ewarn "group only."
+	ewarn "Temp directories:    /var/tmp/${PN}/${SLOT}/default/*"
+	use sqlite && ewarn "SQLite directory:    ${SQLITEDIR}"
+
+	elog
+	elog "The WebDAV CGI config is located at ${WEBDAVCONFIG}."
+	elog
+	elog "An example Apache HTTP server configuration snippet is available in"
+	elog "/usr/share/doc/${PF} in the file ${APACHEEXAMPLECONFIG}"
```
Comment (gktrk) [inline www-apps/webdavcgi/webdavcgi-1.1.1.ebuild]: Better to use `${ROOT%/}/usr/share/doc/${PF}` here

**Comment by gktrk** [comment]
Are there any security implications to the mode 1777 ?

File: dev-perl/DateTime-Format-Human-Duration/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>c.affolter@stepping-stone.ch</email>
+		<name>Christian Affolter</name>
+		<description>Maintainer</description>
```
Comment (paraenggu) [inline dev-perl/DateTime-Format-Human-Duration/metadata.xml]: The `<description />` tag is in the [metadata.xml](https://github.com/gentoo/gentoo/blob/master/www-apps/webdavcgi/metadata.xml) of the old webdavcgi ebuild, that's why I added it here as well. Do you want me to remove it?

File: dev-perl/DateTime-Format-Human-Duration/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	</maintainer>
+	<maintainer type="person">
+		<email>dev-zero@gentoo.org</email>
+		<description>Proxy maintainer</description>
```
Comment (paraenggu) [inline dev-perl/DateTime-Format-Human-Duration/metadata.xml]: Sure, will add the proxy-maint block as well.

File: www-apps/webdavcgi/webdavcgi-1.1.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+EAPI=6
+
+inherit eutils toolchain-funcs versionator
+
+DESCRIPTION="A Perl CGI for accessing and sharing files, or calendar/addressbooks via WebDAV."
+HOMEPAGE="http://webdavcgi.sourceforge.net/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.bz2"
+
+LICENSE="GPL-3"
```
Comment (paraenggu) [inline www-apps/webdavcgi/webdavcgi-1.1.1.ebuild]: You're right, I will adapt that accordingly.

File: www-apps/webdavcgi/webdavcgi-1.1.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+src_compile() {
+	if use suid; then
+		WEBDAVWRAPPERS="webdavwrapper" # Standard UID/GID wrapper
+
+		use afs      && WEBDAVWRAPPERS+=" webdavwrapper-afs"
+		use kerberos && WEBDAVWRAPPERS+=" webdavwrapper-krb"
+
+		export WEBDAVWRAPPERS
+
+	    local wrapper
```
Comment (paraenggu) [inline www-apps/webdavcgi/webdavcgi-1.1.1.ebuild]: No, it's simply a broken indentation, thanks for spotting it.

File: www-apps/webdavcgi/webdavcgi-1.1.1.ebuild (hunk 1 above)
Comment (paraenggu) [inline www-apps/webdavcgi/webdavcgi-1.1.1.ebuild]: OK, will change that as well.

File: www-apps/webdavcgi/webdavcgi-1.1.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+	dev-perl/JSON
+	dev-perl/List-MoreUtils
+	dev-perl/MIME-tools
+	dev-perl/PerlIO-gzip
+	dev-perl/Quota
+	dev-perl/TimeDate
+	dev-perl/URI
+	dev-perl/UUID-Tiny
+	dev-perl/XML-Simple
+	git? ( dev-vcs/git )
```
Comment (paraenggu) [inline www-apps/webdavcgi/webdavcgi-1.1.1.ebuild]: Should be correct, according to the [GIT backend documentation](http://webdavcgi.sourceforge.net/doc.html#git), one has to install the git binary.

**Comment by paraenggu** [comment]
> Are there any security implications to the mode 1777 ?

There might be security implications if one runs other services on the same system or has system users which have direct access to the file system. They might be able to access temporary or deleted files (in case the thrash feature has been enabled), similar to files within `/tmp`. However, that's the only simple way of having a default out-of-the-box working experience after installing the ebuild. The [ewarn messages](https://github.com/paraenggu/gentoo/blob/webdavcgi-1.1.1/www-apps/webdavcgi/webdavcgi-1.1.1.ebuild#L154-L160) are supposed to inform the user about this circumstances and also provide a hardening hint.

**Comment by paraenggu** [comment]
@gktrk Thanks a lot for taking the time to review this PR. Do you prefer new separate commits for addressing the above issues, or shall I squash/rebase the changes with the original commits?

File: dev-perl/DateTime-Format-Human-Duration/metadata.xml (hunk 3 above)
Comment (gktrk) [inline dev-perl/DateTime-Format-Human-Duration/metadata.xml]: If we add the proxy-maint block with dev-zero's permission, you don't need the description tags on yourself and him.

**Comment by gktrk** [comment]
I pinged dev-zero for his approval on IRC. We prefer squashing them.

**Comment by paraenggu** [comment]
@gktrk I've addressed all your comments and rebased/squashed the changes with the original commits.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/e506143...4c4dd65. Thanks!

**Pull request closed without merging**